src/modules/ce/components/ReferralStepCard.tsx

```diff
@@ -51,22 +54,72 @@ const ReferralStepCard: React.FC<Props> = ({ step }) => {
     };
 
     if (status === CeReferralStepStatus.InProgress) {
-      return <ButtonLink {...buttonProps}>View</ButtonLink>;
+      return (
+        <ButtonLink aria-label={`View step: ${name}`} {...buttonProps}>
+          View
+        </ButtonLink>
+      );
     }
 
     if (status === CeReferralStepStatus.Completed) {
       return (
-        <ButtonLink {...buttonProps} color='grayscale'>
+        <ButtonLink
+          aria-label={`View step: ${name}`}
+          {...buttonProps}
+          color='grayscale'
+        >
           View
         </ButtonLink>
       );
     }
-  }, [opportunityId, projectId, referral.id, status, step]);
+  }, [name, opportunityId, projectId, referral.id, status, step]);
+
+  const collapsed = step.status === CeReferralStepStatus.Completed;
+  const locked = step.status === CeReferralStepStatus.Unavailable;
+  const paperSx = useMemo(() => {
+    if (locked)
+      return {
+        color: 'grayscale.main',
+      };
+  }, [locked]);
+
+  const lastUpdatedString = useMemo(() => {
+    const dateString = parseHmisDateString(updatedAt);
+    const at = dateString ? ` ${formatRelativeDateTime(dateString)}` : '';
+    const by = updatedBy ? ` by ${updatedBy.name}` : '';
```

Comment:
This doesn't feel like something that should live in a referral step component. Maybe we need a param on the existing `lastUpdatedBy` to specify whether to use a relative or exact date?